app/src/components/ArtifactTimeline.tsx

```diff
@@ -0,0 +1,112 @@
+import { Center, ScrollArea, Space, Text, Timeline } from '@mantine/core';
+import { Message } from 'tabler-icons-react';
+import { useEffect, useState } from 'react';
+import ArtifactComments from './ArtifactComments';
+
+interface ArtifactCommentProps {
+  date: string | undefined;
+  body: string | undefined;
+  author: string | undefined;
+}
+
+interface ExtensionArray {
+  extensions: undefined | fhir4.Extension[];
+}
+
+interface commentProps {
+  author: string | undefined;
+  date: string | undefined;
+  body: string | undefined;
+  type: string | undefined;
+}
+
+function addArtifactComment({ date, body, author }: ArtifactCommentProps) {
+  return (
+    <div>
+      <ArtifactComments date={date} body={body} author={author} />
+    </div>
+  );
+}
+
+function noCommentsAvailable() {
+  return (
+    <div>
+      <Center>
+        <Text color="red">
+          <i>No Comments Found for Resource</i>{' '}
+        </Text>
+      </Center>
+    </div>
+  );
+}
+
+export default function ArtifactTimeline(extensionArr: ExtensionArray) {
+  const [height, setWindowHeight] = useState(0);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setWindowHeight(window.innerHeight);
+    };
+    handleResize();
+    window.addEventListener('resize', handleResize);
+    return window.removeEventListener('resize', handleResize);
+  }, []);
+
+  if (extensionArr?.extensions) {
+    if (extensionArr.extensions.length === 0) {
+      return <div>{noCommentsAvailable()}</div>;
```

Comment:
Honestly, I don't think you need a function for `noCommentsAvailable()`. I would just return the contents of the function here.

Reply:
I definitely think it wouldn't be necessary to use a function if it was just one instance of the call but since I use it three times I think it helps to get rid of repetitive code :)